src/mpyl/steps/deploy/dagster.py

```diff
@@ -55,12 +43,238 @@ def __evaluate_results(self, results: List[Output]):
+        )
+
+        values_path = Path(step_input.project_execution.project.target_path)
+
+        write_chart(
+            chart={},
+            chart_path=values_path,
+            chart_metadata="",
+            values=user_code_deployment,
+        )
+        return user_code_deployment, values_path
+
+    @staticmethod
+    def add_server_to_server_list(
+        core_api: CoreV1Api, user_code_deployment: dict, dagster_config: DagsterConfig
+    ) -> Output:
+        config_map = get_config_map(
+            core_api,
+            dagster_config.base_namespace,
+            dagster_config.workspace_config_map,
+        )
+        dagster_workspace = yaml.safe_load(
+            config_map.data[dagster_config.workspace_file_key]
+        )
+
+        server_names = [
+            w["grpc_server"]["location_name"] for w in dagster_workspace["load_from"]
+        ]
+
+        # If the server new (not in existing workspace.yml), we append it
+        user_code_name_to_deploy = user_code_deployment["deployments"][0]["name"]
+        if user_code_name_to_deploy not in server_names:
+            dagster_workspace["load_from"].append(
+                to_grpc_server_entry(
+                    host=user_code_name_to_deploy,
+                    port=user_code_deployment["deployments"][0]["port"],
+                    location_name=user_code_name_to_deploy,
+                )
+            )
+            updated_config_map = update_config_map_field(
+                config_map=config_map,
+                field=dagster_config.workspace_file_key,
+                data=dagster_workspace,
+            )
+            return replace_config_map(
+                core_api,
+                dagster_config.base_namespace,
+                dagster_config.workspace_config_map,
+                updated_config_map,
+            )
+        return Output(
+            success=True,
+            message="Server name already exists in list, no addition needed",
+            produced_artifact=None,
+        )
+
+    @staticmethod
+    def restart_dagster_instances(
+        logger, apps_api, dagster_config: DagsterConfig
+    ) -> List[Output]:
+        # restarting ui and daemon
+        rollout_restart_daemon_output = rollout_restart_deployment(
+            logger,
+            apps_api,
+            dagster_config.base_namespace,
+            dagster_config.daemon,
+        )
+
+        if rollout_restart_daemon_output.success:
+            logger.info(rollout_restart_daemon_output.message)
+            rollout_restart_server_output = rollout_restart_deployment(
+                logger,
+                apps_api,
+                dagster_config.base_namespace,
+                dagster_config.webserver,
+            )
+            logger.info(rollout_restart_server_output.message)
+            return [rollout_restart_daemon_output, rollout_restart_server_output]
+        return [rollout_restart_daemon_output]
+
+
+class HelmTemplateDagster(Step, DagsterBase):
```

Comment:
It actually might be nicer to put the 3 steps in its own files and the base class in a utility? This file is a bit of a mess now, would be more clear if it's split I think

Reply:
I agree it'd be nicer, but once we can create the Dagster ConfigMap dynamically, only one of the classes will remain in the end. This is merely an in-between stage where we still need all 3 for migration purposes.

Reply:
Pinging @kgrunert just in case